examples/non-fungible-token/tests/workspaces/test_core.rs

```diff
@@ -185,7 +185,7 @@ async fn simulate_transfer_call_receiver_panics_and_nft_resolve_transfer_produce
             Some("transfer & call"),
             "incorrect message",
         ))
-        .gas(30_000_000_000_000)
+        .gas(1)
```

Comment:
@frol FYI: this fixed the issue that was so confusing.

Somehow the gas amount for this one has changed, I guess it'd be nice for me to investigate how those are calculated. 
There's also an option to supply more gas and still let it fail, but then it manages to emit a log message and fails 
when the assertion happens. 
It seems to me like a somewhat flaky test, perhaps it'd be nice to revisit those in the near future and have some
good docs describing the behaviour.

Reply:
@ruseinov Well, this is totally unexpected that it succeeds as the function call definitely cannot complete within 1 gas limit, especially if before that it could not succeed within 30 Tgas limit. Please, dig more into it.

Reply:
> @ruseinov Well, this is totally unexpected that it succeeds as the function call definitely cannot complete within 1 gas limit, especially if before that it could not succeed within 30 Tgas limit. Please, dig more into it.

The way that it currently works is:
1. If we give it 30 Tgas - it succeeds, because it needs way less
2. If we give it something around 20 - then it fails, but emits a log message, which fails the test.
3. With 1 gas as an extreme it does fail and emits no log messages.

I'll look into why.

Reply:
I'm not sure though what has changed the gas requirements in the first place. And that's what bothers me the most.